### PR TITLE
Fix `RV_RETURN_VALUE_IGNORED_BAD_PRACTICE` SpotBugs violations

### PR DESCRIPTION
**Pull request by basil**
Fixes most `RV_RETURN_VALUE_IGNORED_BAD_PRACTICE`, mostly by switching to NIO.2 where possible.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@StefanSpieker 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/main/java/hudson/model/labels/LabelAtom.java
Comment (basil) [inline core/src/main/java/hudson/model/labels/LabelAtom.java:208]: Needed to avoid errors like this on Windows:

```java
java.nio.file.InvalidPathException: Illegal char <<> at index 84: C:\Jenkins\workspace\Core_jenkins_PR-6036\test\target\j h1212842842721259356\labels\<img\src='x' onerror=alert(123)>xss.xml
	at java.base/sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:182)
	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:153)
	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
	at java.base/sun.nio.fs.WindowsPath.parse(WindowsPath.java:92)
	at java.base/sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:229)
	at java.base/java.io.File.toPath(File.java:2329)
	at hudson.XmlFile.exists(XmlFile.java:247)
	at hudson.model.labels.LabelAtom.load(LabelAtom.java:198)
	at jenkins.model.Jenkins.getLabelAtom(Jenkins.java:2088)
	at jenkins.model.Jenkins.getLabel(Jenkins.java:2066)
	at hudson.model.Label.get(Label.java:601)
```

Since the `InvalidPathException` is expected in many cases, we catch it and log it at `Level.FINE` to avoid creating unnecessary log spam.

**Comment by basil** [comment]
This PR was tested successfully in a [passing Windows CI run](https://ci.jenkins.io/job/Core/job/jenkins/job/PR-6036/3/) in #6036.

**Review by timja — APPROVED** [review]

**Review by StefanSpieker — APPROVED** [review]

**Comment by basil** [comment]
This PR was tested successfully in a [passing BOM run](https://ci.jenkins.io/job/Tools/job/bom/job/PR-757/1/) in jenkinsci/bom#757.

File: core/src/main/java/hudson/model/labels/LabelAtom.java
Comment (jglick) [reply core/src/main/java/hudson/model/labels/LabelAtom.java:208]: I thought I remembered some utility to abstract this, but all I see is https://github.com/jenkinsci/jenkins/blob/f29b4789834a7562b44d8c3a0abcd9fb0c927f27/core/src/main/java/hudson/Util.java#L1717-L1728

File: core/src/main/java/hudson/PluginManager.java
Comment (jglick) [inline core/src/main/java/hudson/PluginManager.java:357]: Do we even need to check that?

File: core/src/main/java/hudson/WebAppMain.java
Comment (jglick) [inline core/src/main/java/hudson/WebAppMain.java:200]: At least adjust the comment, or delete the whole clause? Or rethrow `IOException` as `NoHomeDir`?

File: core/src/main/java/hudson/model/FileParameterValue.java
Comment (jglick) [inline core/src/main/java/hudson/model/FileParameterValue.java:322]: Maybe just log instead? I do not think callers would be expecting an unchecked exception here.

File: core/src/main/java/hudson/model/UpdateCenter.java
Comment (jglick) [inline core/src/main/java/hudson/model/UpdateCenter.java:1887]: Note: behavioral change.

File: core/src/main/java/hudson/util/io/ReopenableRotatingFileOutputStream.java
Comment (jglick) [inline core/src/main/java/hudson/util/io/ReopenableRotatingFileOutputStream.java:76]: Better just log it.

File: core/src/main/java/hudson/util/io/RewindableRotatingFileOutputStream.java
Comment (jglick) [inline core/src/main/java/hudson/util/io/RewindableRotatingFileOutputStream.java:74]: ditto

File: core/src/main/java/jenkins/fingerprints/FileFingerprintStorage.java
Comment (jglick) [inline core/src/main/java/jenkins/fingerprints/FileFingerprintStorage.java:339]: Not sure about this. Depends on the caller.

File: src/spotbugs/spotbugs-excludes.xml
Comment (jglick) [inline src/spotbugs/spotbugs-excludes.xml:490]: I would rather use `@SuppressFBWarnings` with targeted `justification`s.

File: src/spotbugs/spotbugs-excludes.xml
Comment (jglick) [inline src/spotbugs/spotbugs-excludes.xml:708]: Unrelated right? also, #6018

**Review by jglick — COMMENTED** [review]
Thanks—long overdue! Some places I think we wanted to recover gracefully rather than throwing an exception (but silently ignoring the error was bad).

File: core/src/main/java/hudson/model/labels/LabelAtom.java
Comment (basil) [reply core/src/main/java/hudson/model/labels/LabelAtom.java]: Thanks for pointing this out. I used `Util.fileToPath` anywhere that a method was declared to throw `IOException`.

File: core/src/main/java/hudson/PluginManager.java
Comment (basil) [reply core/src/main/java/hudson/PluginManager.java]: No, the original code was unnecessary. I removed the pointless check.

File: core/src/main/java/hudson/model/FileParameterValue.java
Comment (basil) [reply core/src/main/java/hudson/model/FileParameterValue.java:318]: Just a few lines above, we convert `IOException` into `Error`, so I feel like all bets are off with this code path. I made both that path and this one throw `UncheckedIOException`, as at least that seems better than `Error`.

File: core/src/main/java/hudson/model/UpdateCenter.java
Comment (basil) [reply core/src/main/java/hudson/model/UpdateCenter.java:1887]: Yeah, this seemed like more straightforward behavior to me.

File: core/src/main/java/hudson/util/io/ReopenableRotatingFileOutputStream.java
Comment (basil) [reply core/src/main/java/hudson/util/io/ReopenableRotatingFileOutputStream.java]: Done

File: core/src/main/java/hudson/util/io/RewindableRotatingFileOutputStream.java
Comment (basil) [reply core/src/main/java/hudson/util/io/RewindableRotatingFileOutputStream.java]: Done

File: core/src/main/java/jenkins/fingerprints/FileFingerprintStorage.java
Comment (basil) [reply core/src/main/java/jenkins/fingerprints/FileFingerprintStorage.java]: I made this log instead.

File: src/spotbugs/spotbugs-excludes.xml
Comment (basil) [reply src/spotbugs/spotbugs-excludes.xml:456]: This is probably the same false positive as the other two, so all three can be triaged together in the future.

File: src/spotbugs/spotbugs-excludes.xml
Comment (basil) [reply src/spotbugs/spotbugs-excludes.xml]: Yeah, I took care of this in another PR and after merging it's gone from the diff.

File: core/src/main/java/hudson/XmlFile.java
Comment (basil) [inline core/src/main/java/hudson/XmlFile.java:254]: Technically an API change, but there are no plugin consumers of this API, and all the consumers in core were already throwing `IOException` anyway.

File: core/src/main/java/hudson/util/TextFile.java
Comment (basil) [inline core/src/main/java/hudson/util/TextFile.java:62]: Technically an API change, but there are no plugin consumers of this API, and all the consumers in core were already throwing `IOException` anyway.

File: core/src/main/java/hudson/XmlFile.java
Comment (basil) [inline core/src/main/java/hudson/XmlFile.java:251]: Technically an API change that affects 7 plugins at compile-time, but I don't think the addition of the `throws` should affect those seven plugins at runtime. I think this is probably fine, but if people feel strongly otherwise I can convert this to an `UncheckedIOException` in order to preserve the existing method signature for those 7 plugins.

**Review by jglick — APPROVED** [review]
Did not do a very close review but looks good.

File: core/src/main/java/hudson/XmlFile.java
Comment (jglick) [reply core/src/main/java/hudson/XmlFile.java:251]: Which plugins?

Comment (basil) [reply core/src/main/java/hudson/XmlFile.java:251]: @jglick Did you have any opinion about this part? This is what I'm the most unsure about, so would be good to hear if you think this is OK or if you'd prefer the `UncheckedIOException` approach here.

Comment (jglick) [reply core/src/main/java/hudson/XmlFile.java:251]: Probably fine—may cause a compilation error on upgrade, but trivially resolved.

Comment (basil) [reply core/src/main/java/hudson/XmlFile.java:251]: copyartifact
    ecutest
    gerrit-trigger
    google-container-registry-auth
    jqs-monitoring
    nodejs
    nuget
    xvfb

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

File: core/src/main/java/hudson/util/TextFile.java
Comment (timja) [inline core/src/main/java/hudson/util/TextFile.java:99]: caused https://issues.jenkins.io/browse/JENKINS-67624

Comment (jglick) [reply core/src/main/java/hudson/util/TextFile.java:99]: I guess `createDirectories` was meant here?

**Pull request merged**
